Build failure: yabause

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix-build -A yabause

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310505923

### Relevant log output

```console
nix-build -A yabause
this derivation will be built:
  /nix/store/3nazgixh14y5sxff9wlii3b7x7af65qy-yabause-0.9.15.drv
this path will be fetched (2.32 MiB download, 2.37 MiB unpacked):
  /nix/store/y9jh787v3rrh15qc8v8n00lcb7ip7g15-yabause-0.9.15.tar.gz
copying path '/nix/store/y9jh787v3rrh15qc8v8n00lcb7ip7g15-yabause-0.9.15.tar.gz' from 'https://cache.nixos.org'...
building '/nix/store/3nazgixh14y5sxff9wlii3b7x7af65qy-yabause-0.9.15.drv'...
Running phase: qtPreHook
Running phase: unpackPhase
unpacking source archive /nix/store/y9jh787v3rrh15qc8v8n00lcb7ip7g15-yabause-0.9.15.tar.gz
source root is yabause-0.9.15
setting SOURCE_DATE_EPOCH to timestamp 1472067423 of file "yabause-0.9.15/src/yabause.h"
Running phase: patchPhase
applying patch /nix/store/mk5rx53rgx3x6p5k8rfv8a90nh063mb9-linkage-rwx-linux-elf.patch
patching file src/sh2_dynarec/linkage_x64.s
Hunk #1 succeeded at 744 (offset -3 lines).
patching file src/sh2_dynarec/linkage_x86.s
Hunk #1 succeeded at 740 (offset -3 lines).
applying patch /nix/store/d7nh21wja2xczx3z7cjybm40ck7wgaqg-0001-Fixes-for-Qt-5.11-upgrade.patch
patching file src/qt/ui/UICheatRaw.cpp
patching file src/qt/ui/UICheatRaw.h
patching file src/qt/ui/UICheats.cpp
patching file src/qt/ui/UIHexInput.h
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/share/doc/yabause -DCMAKE_INSTALL_INFODIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/include -DCMAKE_INSTALL_SBINDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15 -DYAB_NETWORK=ON -DYAB_OPTIMIZED_DMA=ON -DYAB_PORTS=qt
CMake Warning (dev) at CMakeLists.txt:1 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 14.3.0
-- The CXX compiler identification is GNU 14.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:3 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
error: Cannot build '/nix/store/3nazgixh14y5sxff9wlii3b7x7af65qy-yabause-0.9.15.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/4cx9h4xcd29q7bhv318zl6qyh09i2m9k-yabause-0.9.15
       Last 25 log lines:
       > This warning is for project developers.  Use -Wno-dev to suppress it.
       >
       > -- The C compiler identification is GNU 14.3.0
       > -- The CXX compiler identification is GNU 14.3.0
       > -- Detecting C compiler ABI info
       > -- Detecting C compiler ABI info - done
       > -- Check for working C compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/gcc - skipped
       > -- Detecting C compile features
       > -- Detecting C compile features - done
       > -- Detecting CXX compiler ABI info
       > -- Detecting CXX compiler ABI info - done
       > -- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
       > -- Detecting CXX compile features
       > -- Detecting CXX compile features - done
       > CMake Error at CMakeLists.txt:3 (cmake_minimum_required):
       >   Compatibility with CMake < 3.5 has been removed from CMake.
       >
       >   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
       >   to tell CMake that the project requires at least <min> but has been updated
       >   to work with policies introduced by <max> or earlier.
       >
       >   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run:
         nix log /nix/store/3nazgixh14y5sxff9wlii3b7x7af65qy-yabause-0.9.15.drv
```

### Additional context

_No response_

### System metadata

- system: `"x86_64-linux"`
 - host os: `Linux 6.12.54, NixOS, 25.11 (Xantusia), 25.11.20251022.01f116e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/sqilaxdxljyalv530z1qrd1q3nd4lyxq-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/emulators/yabause/default.nix
+++ b/pkgs/applications/emulators/yabause/default.nix
@@ -18,7 +18,7 @@ mkDerivation rec {
   version = "0.9.15";
 
   src = fetchurl {
-    url = "https://download.tuxfamily.org/yabause/releases/${version}/${pname}-${version}.tar.gz";
+    url = "https://web.archive.org/web/20250129091147/http://free.downloads.tuxfamily.net/yabause/releases/${version}/yabause-${version}.tar.gz";
     sha256 = "1cn2rjjb7d9pkr4g5bqz55vd4pzyb7hg94cfmixjkzzkw0zw8d23";
   };
 
@@ -47,7 +47,15 @@ mkDerivation rec {
     "-DYAB_NETWORK=ON"
     "-DYAB_OPTIMIZED_DMA=ON"
     "-DYAB_PORTS=qt"
+    "-DSH2_DYNAREC=OFF" # https://github.com/Yabause/yabause/issues/270
   ];
+
+  postPatch = ''
+    substituteInPlace {./,src/,src/runner/}CMakeLists.txt \
+      --replace-fail "cmake_minimum_required(VERSION 2.8)" "cmake_minimum_required(VERSION 3.10)"
+    substituteInPlace src/{c68k,musashi}/CMakeLists.txt \
+      --replace-fail "cmake_minimum_required(VERSION 2.6)" "cmake_minimum_required(VERSION 3.10)"
+  '';
 
   meta = with lib; {
     description = "Open-source Sega Saturn emulator";

